src/Analyzers/CSharp/Analyzers/RemoveUnnecessaryImports/CSharpRemoveUnnecessaryImportsDiagnosticAnalyzer.cs

```diff
@@ -15,22 +17,34 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.RemoveUnnecessaryImports
 {
+#if CODE_STYLE
+    using Resources = CSharpCodeStyleResources;
+#else
+    using Resources = CSharpFeaturesResources;
+#endif
+
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     internal sealed class CSharpRemoveUnnecessaryImportsDiagnosticAnalyzer :
         AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer
     {
-        private static readonly LocalizableString s_TitleAndMessageFormat =
-            new LocalizableResourceString(nameof(CSharpFeaturesResources.Using_directive_is_unnecessary), CSharpFeaturesResources.ResourceManager, typeof(CSharpFeaturesResources));
+        private static readonly LocalizableString s_titleAndMessageFormat =
+            new LocalizableResourceString(nameof(Resources.Using_directive_is_unnecessary), Resources.ResourceManager, typeof(Resources));
 
         protected override LocalizableString GetTitleAndMessageFormatForClassificationIdDescriptor()
-            => s_TitleAndMessageFormat;
+            => s_titleAndMessageFormat;
 
         // C# has no need to do any merging of using statements.  Only VB needs to
         // merge import clauses to an import statement if it all the import clauses
         // are unnecessary.
         protected override ImmutableArray<SyntaxNode> MergeImports(ImmutableArray<SyntaxNode> unnecessaryImports)
             => unnecessaryImports;
 
+        protected override IUnnecessaryImportsProvider UnnecessaryImportsProvider
+            => CSharpUnnecessaryImportsProvider.Instance;
+
+        protected override bool IsRegularCommentOrDocComment(SyntaxTrivia trivia)
+            => trivia.IsRegularComment() || trivia.IsDocComment();
```

Comment:
why?

Reply:
Language services are not accessible from analyzers, so analyzer implementation have to fall back to directly invoking extension methods. We may want to consider rewriting important language services such as `ISyntaxFactsService` to be implemented as a utility in a workspace independent way and then a thin language service wrapper in Workspace layer. For now, this was the only place in the 5 analyzers that I attempted to port that used a language service, so I took this route. Note that code fixes can still discover and access language services, so they are unaffected.

Reply:
> to be implemented as a utility in a workspace independent way 

Note: it already has that capability.  You don't need discovery for it.  i.e. you can reference CSharpSyntaxFactsService.Instance and VisualBasicSyntaxFactsService.Instance.  Subclasses can just return these directly in analyzers so we don't need any workspace info.

Reply:
That will not work as these types still implement `ILanguageService` which is defined in MS.CA.Workspaces. I was suggesting that any service that needs to be accesible from analyzer should be split into a pure utility (with likely singleton instances) and then a thin wrapper language service that delegates to the same utility to allow MEF based language service discovery in layers above that do have access to Workspaces and `ILanguageService`. That was the precise change I did in https://github.com/dotnet/roslyn/pull/41363/commits/a6579cca87122296b0cdee2c1eb931f20b4fafbd for remove unnecessary imports analyzer.

Reply:
easy solutoin:

FileA.cs
```
internal partial class CSharpSyntaxFactsService { ... impl ... }
```

FileB.cs

```
internal partial class CSharpSyntaxFactsService : ILanguageService { ... empty ... }
```

:)

Reply:
> I was suggesting that any service that needs to be accesible from analyzer should be split into a pure utility 

yup.  agreed.  i would just have us to that *immediately* (in another PR) so it's ready for this PR.

Reply:
can this be unresolved.  with my PR this should be possible to support without having to change the feature code.